src/convertToObject.js

```diff
@@ -3,16 +3,38 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const propertiesArray = sourceString.split(';');
+  const result = {};
+
+  propertiesArray.forEach(property => {
+    const arr = property.split(':');
+    let nameOfProperty = arr[0].split(' ').join();
+
+    do {
+      nameOfProperty = nameOfProperty.replace(',', '');
+    } while (nameOfProperty.includes(','));
+
+    do {
+      nameOfProperty = nameOfProperty.replace('\n', '');
+    } while (nameOfProperty.includes('\n'));
```

Comment:
this logic is too complicated. you can just trim key and value and check if both of them exist after